server/api/types.go

```diff
@@ -45,6 +62,23 @@ type Result struct {
 	Line          string   `json:"line"`
 }
 
+type ResultV2 struct {
+	Tree    string        `json:"tree"`
+	Version string        `json:"version"`
+	Path    string        `json:"path"`
+	Lines   []*ResultLine `json:"lines"`
+	// Will never be sent over wire, used to deduplicate
+	ContextLines map[int]*ResultLine `json:"-"`
+}
+
+type ResultLine struct {
+	LineNumber int `json:"lno"`
+	// Bounds may or may not be defined. If they are,
+	// then this line is a match. Otherwise it's contex
```

Comment:
This will need to be revisited (I think) once we start checking context lines for matches (which we don't do right now!)